### PR TITLE
mon/MDSMonitor: fix wrong positive result of jewel flag check

### PR DESCRIPTION
**Pull request by xiexingguo**

### CODE REVIEW

**Comment by xiexingguo** [comment]
retest this please

**Comment by xiexingguo** [comment]
@jcsp Some fs-related fixes, please help to review. Thanks.

File: src/mon/MDSMonitor.cc [hunk 1]
```diff
@@ -2385,16 +2390,14 @@ int MDSMonitor::legacy_filesystem_command(
...
       return -EINVAL;
     }
 
     const MDSMap& mdsmap =
       pending_fsmap.filesystems.at(pending_fsmap.legacy_client_fscid)->mds_map;
       
-    if (!mdsmap.allows_multimds() &&
-	maxmds > mdsmap.get_max_mds() &&
-	maxmds > 1) {
+    if (!mdsmap.allows_multimds() && maxmds > 1) {
```
Comment (gregsfortytwo) [inline src/mon/MDSMonitor.cc]: Nope. We need to deal here with upgraded clusters that were set up with multimds prior to the lockout.

File: src/mon/MDSMonitor.cc [hunk 2]
```diff
@@ -2385,16 +2390,14 @@ int MDSMonitor::legacy_filesystem_command(
   if (prefix == "mds set_max_mds") {
     // NOTE: deprecated by "fs set max_mds"
     int64_t maxmds;
-    if (!cmd_getval(g_ceph_context, cmdmap, "maxmds", maxmds) || maxmds < 0) {
+    if (!cmd_getval(g_ceph_context, cmdmap, "maxmds", maxmds) || maxmds <= 0) {
```
Comment (gregsfortytwo) [inline src/mon/MDSMonitor.cc]: I believe you need to be able to set maxmds 0 to do a proper shutdown and cleanup of the FS?

File: src/mon/MDSMonitor.cc [hunk 3]
```diff
@@ -1815,8 +1815,7 @@ class SetHandler : public FileSystemCommandHandler
         return -EINVAL;
       }
 
-      if (!fs->mds_map.allows_multimds() && n > fs->mds_map.get_max_mds() &&
```
Comment (jcsp) [inline src/mon/MDSMonitor.cc]: The problem with this change is in the case where someone has:

1.  setallow_multimds tru
2.  set max_mds to something >1
3.  set allow_multimds false

In this situation, we should allow people to decrease max_mds for their filesystem.

So I think just drop this commit?

**Comment by jcsp** [comment]
This all looks good apart from the "drop weird limit for multimds checking" commit, which I think we should drop.

File: src/mon/MDSMonitor.cc (hunk 2 above)
Comment (jcsp) [inline src/mon/MDSMonitor.cc]: No, you can just do cluster_down, fail the ranks, and then "fs rm".  I don't think there's any gain to formally deactivating rank 0 if you're about to destroy the FS.

**Comment by xiexingguo** [comment]
@jcsp @gregsfortytwo Thank you both for the quick review. Dropped that change.

**Comment by jcsp** [comment]
Passing here: http://pulpito.ceph.com/jspray-2016-06-02_01:58:41-fs-wip-jcsp-testing-20160601---basic-mira/

**Comment by jcsp** [comment]
Created backport ticket for the bug fix parts of this http://tracker.ceph.com/issues/16136

**Pull request merged**